src/Microsoft.PowerShell.ConsoleGuiTools/ConsoleGui.cs

```diff
@@ -56,9 +58,54 @@ public void Start(ApplicationData applicationData)
             top.Add(menu);
 
             var gridHeaders = applicationData.DataTable.DataColumns.Select((c) => c.Label).ToList();
-            // We add one as the offset here to line it up with the data.
-            win.Add(new Label(GetPaddedString(gridHeaders, top.Frame.Width - 3, offset: 1)));
-            
+            var columnWidths = new int[gridHeaders.Count];
+            for (int i = 0; i < gridHeaders.Count; i++)
+            {
+                columnWidths[i] = gridHeaders[i].Length;
+            }
+
+            // calculate the width of each column based on longest string in each column for each row
+            foreach (var row in applicationData.DataTable.Data)
+            {
+                int index = 0;
+                foreach (var col in row.Values.Take(NumberOfObjectsToInspect))
```

Comment:
wait actually... we can use:

```
top.Frame.Height
```

then we'll be sure that everything you initially see will be fitted as nice as possible.